src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -884,18 +884,9 @@ public BoundStatement Switch(BoundExpression ex, ImmutableArray<SyntheticSwitchS
             statements.Add(null); // placeholder at statements[0] for the dispatch
             foreach (var section in sections)
             {
-                LabelSymbol sectionLabel;
-                // If the section only contains a goto statement, use that label.
```

Comment:
IL generator should be able to do this optimization as a part of branch folding pass and it will check whether this is safe to do.
I assume most of the IL diffs are in Debug tests.